capsules/src/process_console.rs

```diff
@@ -183,12 +256,458 @@ impl<'a, C: ProcessManagementCapability> ProcessConsole<'a, C> {
                 self.rx_in_progress.set(true);
                 let _ = self.uart.receive_buffer(buffer, 1);
                 self.running.set(true);
-                //debug!("Starting process console");
             });
+            //Starts the process console while printing base information about
+            //The kernel version installed
+            let mut console_writer = ConsoleWriter::new();
+            let _ = write(
+                &mut console_writer,
+                format_args!(
+                    "Kernel version: {}\r\n",
+                    option_env!("TOCK_KERNEL_VERSION").unwrap_or("unknown")
+                ),
+            );
+            let _ = self.write_bytes(&(console_writer.buf)[..console_writer.size]);
+            let _ = self.write_bytes(b"Welcome to the process console.\n");
+            let _ = self.write_bytes(
+                b"Valid commands are: help status list stop start fault process kernel\n",
+            );
         }
         Ok(())
     }
+    //simple state machine that identifies the next state
+    fn next_state(&self, state: WriterState) -> WriterState {
+        match state {
+            WriterState::KernelStart => WriterState::KernelBss,
+            WriterState::KernelBss => WriterState::KernelInit,
+            WriterState::KernelInit => WriterState::KernelStack,
+            WriterState::KernelStack => WriterState::KernelRoData,
+            WriterState::KernelRoData => WriterState::KernelText,
+            WriterState::KernelText => WriterState::Empty,
+            WriterState::ProcessStart => WriterState::ProcessGrant,
+            WriterState::ProcessGrant => WriterState::ProcessHeapUnused,
+            WriterState::ProcessHeapUnused => WriterState::ProcessHeap,
+            WriterState::ProcessHeap => WriterState::ProcessData,
+            WriterState::ProcessData => WriterState::ProcessStack,
+            WriterState::ProcessStack => WriterState::ProcessStackUnused,
+            WriterState::ProcessStackUnused => WriterState::ProcessFlash,
+            WriterState::ProcessFlash => WriterState::ProcessProtected,
+            WriterState::ProcessProtected => WriterState::Empty,
+            WriterState::Empty => WriterState::Empty,
+        }
+    }
+    //defines the behavior at each state
+    fn create_state_buffer(&self, state: WriterState, process_id: Option<ProcessId>) {
+        match state {
+            WriterState::KernelBss => {
+                let mut console_writer = ConsoleWriter::new();
+                let kernel_info = KernelInfo::new(self.kernel);
+                let init_bottom: usize = kernel_info.get_kernel_init_end() as usize;
+                //let bss_start: usize = kernel_info.get_kernel_bss_start() as usize;
```

Comment:
I just want to check if I'm missing something, why isn't bss_start used here?

Reply:
init_bottom is the same as bss_start so you only need one of them.  I only created both functions for the sake of completeness.